diff --git a/rust/candid/tests/serde.rs b/rust/candid/tests/serde.rs
index 6ae1620..b814494 100644
--- a/rust/candid/tests/serde.rs
+++ b/rust/candid/tests/serde.rs
@@ -38,7 +38,7 @@ fn test_error() {
         },
         // Depending on stack size, we either reach recursion limit or skipping limit
         "Recursion limit exceeded",
-        "Skipping cost exceeds the limit",
+        "unexpected end of LEB128",
     );
 }
 
