diff --git a/Cargo.lock b/Cargo.lock
index a50bd57..6047a0d 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2455,8 +2455,8 @@ dependencies = [
  "hex",
  "http 1.2.0",
  "http-body 1.0.1",
- "ic-certification 3.0.2",
- "ic-transport-types 0.39.3",
+ "ic-certification",
+ "ic-transport-types",
  "ic-verify-bls-signature",
  "k256 0.13.4",
  "leb128",
@@ -2648,18 +2648,6 @@ dependencies = [
  "slotmap",
 ]
 
-[[package]]
-name = "ic-certification"
-version = "2.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "e64ee3d8b6e81b51f245716d3e0badb63c283c00f3c9fb5d5219afc30b5bf821"
-dependencies = [
- "hex",
- "serde",
- "serde_bytes",
- "sha2 0.10.8",
-]
-
 [[package]]
 name = "ic-certification"
 version = "3.0.2"
@@ -3111,7 +3099,7 @@ dependencies = [
  "ic-cdk 0.16.0",
  "ic-cdk-macros 0.9.0",
  "ic-cdk-timers",
- "ic-certification 3.0.2",
+ "ic-certification",
  "ic-crypto-tree-hash",
  "ic-icrc1",
  "ic-icrc1-tokens-u64",
@@ -4023,23 +4011,6 @@ dependencies = [
  "wsl",
 ]
 
-[[package]]
-name = "ic-transport-types"
-version = "0.37.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "875dc4704780383112e8e8b5063a1b98de114321d0c7d3e7f635dcf360a57fba"
-dependencies = [
- "candid",
- "hex",
- "ic-certification 2.6.0",
- "leb128",
- "serde",
- "serde_bytes",
- "serde_repr",
- "sha2 0.10.8",
- "thiserror 1.0.69",
-]
-
 [[package]]
 name = "ic-transport-types"
 version = "0.39.3"
@@ -4048,7 +4019,7 @@ checksum = "979ee7bee5a67150a4c090fb012c93c294a528b4a867bad9a15cc6d01cb4227f"
 dependencies = [
  "candid",
  "hex",
- "ic-certification 3.0.2",
+ "ic-certification",
  "leb128",
  "serde",
  "serde_bytes",
@@ -5063,7 +5034,7 @@ dependencies = [
  "ic-icrc1-ledger",
  "ic-sns-cli",
  "ic-utils 0.39.3",
- "pocket-ic 6.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "pocket-ic 6.0.0 (git+https://github.com/dfinity/ic?rev=760e1f764b56f4f655a09789c245da487eccc5cb)",
  "reqwest 0.11.27",
  "rust_decimal",
  "serde",
@@ -5548,14 +5519,14 @@ checksum = "953ec861398dccce10c670dfeaf3ec4911ca479e9c02154b3a215178c5f566f2"
 [[package]]
 name = "pocket-ic"
 version = "6.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "124a2380ca6f557adf8b02517cbfd2f564113230e14cda6f6aadd3dfe156293c"
+source = "git+https://github.com/dfinity/ic?rev=2dd2ccee9679aaf23cf57daaa703c46f7962c29f#2dd2ccee9679aaf23cf57daaa703c46f7962c29f"
 dependencies = [
+ "backoff",
  "base64 0.13.1",
  "candid",
  "hex",
- "ic-certification 2.6.0",
- "ic-transport-types 0.37.1",
+ "ic-certification",
+ "ic-transport-types",
  "reqwest 0.12.12",
  "schemars",
  "serde",
@@ -5566,7 +5537,7 @@ dependencies = [
  "slog",
  "strum",
  "strum_macros",
- "thiserror 1.0.69",
+ "thiserror 2.0.11",
  "tokio",
  "tracing",
  "tracing-appender",
@@ -5577,14 +5548,14 @@ dependencies = [
 [[package]]
 name = "pocket-ic"
 version = "6.0.0"
-source = "git+https://github.com/dfinity/ic?rev=2dd2ccee9679aaf23cf57daaa703c46f7962c29f#2dd2ccee9679aaf23cf57daaa703c46f7962c29f"
+source = "git+https://github.com/dfinity/ic?rev=760e1f764b56f4f655a09789c245da487eccc5cb#760e1f764b56f4f655a09789c245da487eccc5cb"
 dependencies = [
  "backoff",
  "base64 0.13.1",
  "candid",
  "hex",
- "ic-certification 3.0.2",
- "ic-transport-types 0.39.3",
+ "ic-certification",
+ "ic-transport-types",
  "reqwest 0.12.12",
  "schemars",
  "serde",
diff --git a/Cargo.toml b/Cargo.toml
index 426db52..1da5263 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,3 +47,6 @@ serde_json = "1.0.79"
 [profile.dist]
 inherits = "release"
 lto = "thin"
+
+[patch.crates-io]
+pocket-ic = { git = "https://github.com/dfinity/ic", rev = "760e1f764b56f4f655a09789c245da487eccc5cb" }
diff --git a/extensions/nns/Cargo.toml b/extensions/nns/Cargo.toml
index 01bb9d9..798a31e 100644
--- a/extensions/nns/Cargo.toml
+++ b/extensions/nns/Cargo.toml
@@ -6,7 +6,7 @@ edition.workspace = true
 license.workspace = true
 repository.workspace = true
 publish.workspace = true
-build = "build.rs"
+
 
 [build-dependencies]
 dfx-extensions-utils.workspace = true
diff --git a/extensions/nns/build.rs b/extensions/nns/build.rs
deleted file mode 100644
index 0c0a988..0000000
--- a/extensions/nns/build.rs
+++ /dev/null
@@ -1,36 +0,0 @@
-use std::env;
-use std::path::PathBuf;
-
-const REPLICA_REV: &str = "760e1f764b56f4f655a09789c245da487eccc5cb";
-
-const BINARY_DEPENDENCIES: &[(&str, &str)] = &[
-    // (downloaded binary name, renamed binary name)
-    ("ic-nns-init", "ic-nns-init"),
-    ("ic-admin", "ic-admin"),
-    ("sns", "sns-cli"),
-];
-
-fn main() {
-    // keep copy of the dependency in the root of the project, so that cargo-dist will be able to package it into a tarball
-    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    // and also in `target/debug` or `target/release` for development purposes (e.g. cargo run), this is a bit hacky: https://github.com/rust-lang/cargo/issues/9661
-    let target_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap())
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .to_path_buf();
-    for (binary_name, renamed_binary_name) in BINARY_DEPENDENCIES {
-        let bin_in_manifest_dir = manifest_dir.join(renamed_binary_name);
-        let bin_in_target_dir = target_dir.join(renamed_binary_name);
-        dbg!(&bin_in_manifest_dir, &bin_in_target_dir);
-        dfx_extensions_utils::download_ic_binary(REPLICA_REV, binary_name, &bin_in_manifest_dir);
-        if bin_in_target_dir.exists() {
-            std::fs::remove_file(&bin_in_target_dir).unwrap();
-        }
-        std::fs::create_dir_all(&target_dir).unwrap();
-        std::fs::copy(&bin_in_manifest_dir, &bin_in_target_dir).unwrap();
-    }
-}
diff --git a/extensions/sns/Cargo.toml b/extensions/sns/Cargo.toml
index 511e851..2a529ec 100644
--- a/extensions/sns/Cargo.toml
+++ b/extensions/sns/Cargo.toml
@@ -6,7 +6,7 @@ edition.workspace = true
 license.workspace = true
 repository.workspace = true
 publish.workspace = true
-build = "build.rs"
+
 
 [build-dependencies]
 dfx-extensions-utils.workspace = true
diff --git a/extensions/sns/build.rs b/extensions/sns/build.rs
deleted file mode 100644
index de3217c..0000000
--- a/extensions/sns/build.rs
+++ /dev/null
@@ -1,34 +0,0 @@
-use std::env;
-use std::path::PathBuf;
-
-const REPLICA_REV: &str = "760e1f764b56f4f655a09789c245da487eccc5cb";
-
-const BINARY_DEPENDENCIES: &[(&str, &str)] = &[
-    // (downloaded binary name, renamed binary name)
-    ("sns", "sns-cli"),
-];
-
-fn main() {
-    // keep copy of the dependency in the root of the project, so that cargo-dist will be able to package it into a tarball
-    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    // and also in `target/debug` or `target/release` for development purposes (e.g. cargo run), this is a bit hacky: https://github.com/rust-lang/cargo/issues/9661
-    let target_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap())
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .to_path_buf();
-    for (binary_name, renamed_binary_name) in BINARY_DEPENDENCIES {
-        let bin_in_manifest_dir = manifest_dir.join(renamed_binary_name);
-        let bin_in_target_dir = target_dir.join(renamed_binary_name);
-        dbg!(&bin_in_manifest_dir, &bin_in_target_dir);
-        dfx_extensions_utils::download_ic_binary(REPLICA_REV, binary_name, &bin_in_manifest_dir);
-        if bin_in_target_dir.exists() {
-            std::fs::remove_file(&bin_in_target_dir).unwrap();
-        }
-        std::fs::create_dir_all(&target_dir).unwrap();
-        std::fs::copy(&bin_in_manifest_dir, &bin_in_target_dir).unwrap();
-    }
-}
