diff --git a/Cargo.lock b/Cargo.lock
index 60621d7..f5dc3db 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3652,18 +3652,6 @@ dependencies = [
  "tree-deserializer",
 ]
 
-[[package]]
-name = "ic-certification"
-version = "2.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "e64ee3d8b6e81b51f245716d3e0badb63c283c00f3c9fb5d5219afc30b5bf821"
-dependencies = [
- "hex",
- "serde",
- "serde_bytes",
- "sha2 0.10.9",
-]
-
 [[package]]
 name = "ic-certification"
 version = "3.0.2"
@@ -6615,19 +6603,20 @@ dependencies = [
 
 [[package]]
 name = "ic-transport-types"
-version = "0.37.1"
+version = "0.39.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "875dc4704780383112e8e8b5063a1b98de114321d0c7d3e7f635dcf360a57fba"
+checksum = "979ee7bee5a67150a4c090fb012c93c294a528b4a867bad9a15cc6d01cb4227f"
 dependencies = [
  "candid",
  "hex",
- "ic-certification 2.6.0",
+ "ic-certification 3.0.2",
  "leb128",
  "serde",
  "serde_bytes",
+ "serde_cbor",
  "serde_repr",
  "sha2 0.10.9",
- "thiserror 1.0.69",
+ "thiserror 2.0.12",
 ]
 
 [[package]]
@@ -8670,14 +8659,14 @@ dependencies = [
 [[package]]
 name = "pocket-ic"
 version = "6.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "124a2380ca6f557adf8b02517cbfd2f564113230e14cda6f6aadd3dfe156293c"
+source = "git+https://github.com/dfinity/ic?rev=760e1f764b56f4f655a09789c245da487eccc5cb#760e1f764b56f4f655a09789c245da487eccc5cb"
 dependencies = [
+ "backoff",
  "base64 0.13.1",
  "candid",
  "hex",
- "ic-certification 2.6.0",
- "ic-transport-types 0.37.1",
+ "ic-certification 3.0.2",
+ "ic-transport-types 0.39.3",
  "reqwest 0.12.20",
  "schemars",
  "serde",
@@ -8688,7 +8677,7 @@ dependencies = [
  "slog",
  "strum",
  "strum_macros",
- "thiserror 1.0.69",
+ "thiserror 2.0.12",
  "tokio",
  "tracing",
  "tracing-appender",
diff --git a/Cargo.toml b/Cargo.toml
index 3c3787e..c9bd301 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,3 +50,6 @@ ic-sns-cli        = { git = "https://github.com/dfinity/ic", rev = "02571e8215fa
 [profile.dist]
 inherits = "release"
 lto = "thin"
+
+[patch.crates-io]
+pocket-ic = { git = "https://github.com/dfinity/ic", rev = "760e1f764b56f4f655a09789c245da487eccc5cb" }
diff --git a/extensions/nns/Cargo.toml b/extensions/nns/Cargo.toml
index 48c86f2..690c55d 100644
--- a/extensions/nns/Cargo.toml
+++ b/extensions/nns/Cargo.toml
@@ -6,7 +6,7 @@ edition.workspace = true
 license.workspace = true
 repository.workspace = true
 publish.workspace = true
-build = "build.rs"
+
 
 [build-dependencies]
 dfx-extensions-utils.workspace = true
diff --git a/extensions/nns/build.rs b/extensions/nns/build.rs
deleted file mode 100644
index 4b2a73e..0000000
--- a/extensions/nns/build.rs
+++ /dev/null
@@ -1,39 +0,0 @@
-use dfx_extensions_utils::dependencies::dfx::NNS_SNS_REPLICA_REV;
-use std::env;
-use std::path::PathBuf;
-
-const BINARY_DEPENDENCIES: &[(&str, &str)] = &[
-    // (downloaded binary name, renamed binary name)
-    ("ic-nns-init", "ic-nns-init"),
-    ("ic-admin", "ic-admin"),
-    ("sns", "sns-cli"),
-];
-
-fn main() {
-    // keep copy of the dependency in the root of the project, so that cargo-dist will be able to package it into a tarball
-    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    // and also in `target/debug` or `target/release` for development purposes (e.g. cargo run), this is a bit hacky: https://github.com/rust-lang/cargo/issues/9661
-    let target_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap())
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .to_path_buf();
-    for (binary_name, renamed_binary_name) in BINARY_DEPENDENCIES {
-        let bin_in_manifest_dir = manifest_dir.join(renamed_binary_name);
-        let bin_in_target_dir = target_dir.join(renamed_binary_name);
-        dbg!(&bin_in_manifest_dir, &bin_in_target_dir);
-        dfx_extensions_utils::download_ic_binary(
-            NNS_SNS_REPLICA_REV,
-            binary_name,
-            &bin_in_manifest_dir,
-        );
-        if bin_in_target_dir.exists() {
-            std::fs::remove_file(&bin_in_target_dir).unwrap();
-        }
-        std::fs::create_dir_all(&target_dir).unwrap();
-        std::fs::copy(&bin_in_manifest_dir, &bin_in_target_dir).unwrap();
-    }
-}
diff --git a/extensions/sns/Cargo.toml b/extensions/sns/Cargo.toml
index 9ef8ae2..d6f65f6 100644
--- a/extensions/sns/Cargo.toml
+++ b/extensions/sns/Cargo.toml
@@ -6,7 +6,7 @@ edition.workspace = true
 license.workspace = true
 repository.workspace = true
 publish.workspace = true
-build = "build.rs"
+
 
 [build-dependencies]
 dfx-extensions-utils.workspace = true
diff --git a/extensions/sns/build.rs b/extensions/sns/build.rs
deleted file mode 100644
index 01870ab..0000000
--- a/extensions/sns/build.rs
+++ /dev/null
@@ -1,37 +0,0 @@
-use dfx_extensions_utils::dependencies::dfx::NNS_SNS_REPLICA_REV;
-use std::env;
-use std::path::PathBuf;
-
-const BINARY_DEPENDENCIES: &[(&str, &str)] = &[
-    // (downloaded binary name, renamed binary name)
-    ("sns", "sns-cli"),
-];
-
-fn main() {
-    // keep copy of the dependency in the root of the project, so that cargo-dist will be able to package it into a tarball
-    let manifest_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
-    // and also in `target/debug` or `target/release` for development purposes (e.g. cargo run), this is a bit hacky: https://github.com/rust-lang/cargo/issues/9661
-    let target_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap())
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .parent()
-        .unwrap()
-        .to_path_buf();
-    for (binary_name, renamed_binary_name) in BINARY_DEPENDENCIES {
-        let bin_in_manifest_dir = manifest_dir.join(renamed_binary_name);
-        let bin_in_target_dir = target_dir.join(renamed_binary_name);
-        dbg!(&bin_in_manifest_dir, &bin_in_target_dir);
-        dfx_extensions_utils::download_ic_binary(
-            NNS_SNS_REPLICA_REV,
-            binary_name,
-            &bin_in_manifest_dir,
-        );
-        if bin_in_target_dir.exists() {
-            std::fs::remove_file(&bin_in_target_dir).unwrap();
-        }
-        std::fs::create_dir_all(&target_dir).unwrap();
-        std::fs::copy(&bin_in_manifest_dir, &bin_in_target_dir).unwrap();
-    }
-}
