diff --git a/rs/http_endpoints/public/Cargo.toml b/rs/http_endpoints/public/Cargo.toml
index 419dd6413f..acbe354f32 100644
--- a/rs/http_endpoints/public/Cargo.toml
+++ b/rs/http_endpoints/public/Cargo.toml
@@ -27,7 +27,6 @@ ic-interfaces-state-manager = { path = "../../interfaces/state_manager" }
 ic-logger = { path = "../../monitoring/logger" }
 ic-metrics = { path = "../../monitoring/metrics" }
 ic-pprof = { path = "../../monitoring/pprof" }
-ic-backtrace = { path = "../../monitoring/backtrace" }
 ic-registry-client-helpers = { path = "../../registry/helpers" }
 ic-registry-provisional-whitelist = { path = "../../registry/provisional_whitelist" }
 ic-registry-subnet-type = { path = "../../registry/subnet_type" }
@@ -52,6 +51,9 @@ tokio-io-timeout = "1.2.0"
 tower = { version = "0.4.8", features = ["load-shed", "limit", "steer"] }
 url = "2.1.1"
 
+[target.'cfg(target_os = "linux")'.dependencies]
+ic-backtrace = { path = "../../monitoring/backtrace" }
+
 [dev-dependencies]
 bytes = "1.0.1"
 ic-agent = { workspace = true }
diff --git a/rs/replica/Cargo.toml b/rs/replica/Cargo.toml
index 9fc4ffc994..8ab45d26f9 100644
--- a/rs/replica/Cargo.toml
+++ b/rs/replica/Cargo.toml
@@ -44,7 +44,6 @@ ic-registry-local-store = { path = "../registry/local_store" }
 ic-registry-subnet-type = { path = "../registry/subnet_type" }
 ic-replica-setup-ic-network =  { path = "./setup_ic_network" }
 ic-replicated-state = { path = "../replicated_state" }
-ic-backtrace = { path = "../monitoring/backtrace" }
 ic-state-manager = { path = "../state_manager" }
 ic-sys = { path = "../sys" }
 ic-types = { path = "../types/types" }
@@ -73,6 +72,9 @@ tokio = { version = "1.15.0", features = ["full"] }
 tracing = "0.1.13"
 tower = { version = "0.4.11", features = ["util"] }
 
+[target.'cfg(target_os = "linux")'.dependencies]
+ic-backtrace = { path = "../monitoring/backtrace" }
+
 [dev-dependencies]
 assert_cmd = "0.12"
 canister-test = { path = "../rust_canisters/canister_test" }
