diff --git a/rs/nns/handlers/lifeline/impl/lifeline.mo b/rs/nns/handlers/lifeline/impl/lifeline.mo
index 76a4dbb2cd..017dc1234b 100644
--- a/rs/nns/handlers/lifeline/impl/lifeline.mo
+++ b/rs/nns/handlers/lifeline/impl/lifeline.mo
@@ -4,9 +4,9 @@ import Prim "mo:prim";
 import Root "canister:root";
 import Governance "canister:governance";
 
-actor {
-    private let governanceCanister : Principal = Prim.principalOfActor Governance;
-    private let root : Principal = Prim.principalOfActor Root;
+persistent actor {
+    private transient let governanceCanister : Principal = Prim.principalOfActor Governance;
+    private transient let root : Principal = Prim.principalOfActor Root;
 
     type UpgradeRootProposalPayload = {
       wasm_module : Blob;
@@ -36,7 +36,7 @@ actor {
 
     // IC00 is the management canister. We rely on it for the four
     // fundamental methods as listed below.
-    private let ic00 = actor "aaaaa-aa" : actor {
+    private transient let ic00 = actor "aaaaa-aa" : actor {
       install_code : {
         mode : { #install; #reinstall; #upgrade };
         canister_id : Principal;
