diff --git a/rs/nns/sns-wasm/Cargo.toml b/rs/nns/sns-wasm/Cargo.toml
index e3bceada6f..1265591843 100644
--- a/rs/nns/sns-wasm/Cargo.toml
+++ b/rs/nns/sns-wasm/Cargo.toml
@@ -34,7 +34,6 @@ ic-sns-init = { path = "../../sns/init" }
 ic-sns-root = { path = "../../sns/root" }
 ic-types = { path = "../../types/types" }
 ic-utils = { path = "../../utils" }
-ic-wasm = "^0.7.1"
 icrc-ledger-types = { path = "../../../packages/icrc-ledger-types" }
 libflate = "1.1.2"
 maplit = "1.0.2"
@@ -43,6 +42,12 @@ serde_bytes = { workspace = true }
 serde_json = { workspace = true }
 prost = { workspace = true }
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+ic-wasm = { version = "^0.7.1", default_features = false, features = [] }
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+ic-wasm = { version = "^0.7.1" }
+
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
 registry-canister = { path = "../../registry/canister" }
 ic-registry-keys = { path = "../../registry/keys" }
